theano/tensor/basic.py

```diff
@@ -4219,6 +4219,12 @@ def roll(x, shift, axis=None):
         else:
             axis = 0
 
+    if axis < 0:
+        if x.ndim == 0:
+            axis = 0
+        else:
+            axis = axis % x.ndim
```

Comment:
Thanks for your advice, it doesn't work for numpy in that case. I have changed it according to your suggestion.